src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs

```diff
@@ -62,24 +78,19 @@ public void ClientAsyncAuthenticate_EachProtocol_Success()
             }
         }
 
-        [Fact]
-        public void ClientAsyncAuthenticate_MismatchProtocols_Fails()
+        [Theory]
+        [MemberData("ProtocolMismatchData")]
+        public void ClientAsyncAuthenticate_MismatchProtocols_Fails(SslProtocols server, SslProtocols client)
         {
-            foreach (SslProtocols serverProtocol in s_eachSslProtocol)
+            if ((server.ToString() == "Tls12" && new[] { "Ssl3", "Tls", "Tls11" }.Contains(client.ToString()))
+                || (server.ToString() == "Tls11" && new[] { "Ssl3", "Tls" }.Contains(client.ToString()))
+                || (server.ToString() == "Tls" && client.ToString() == "Ssl3"))
```

Comment:
If this aims to be a data-driven test, then you shouldn't have the ProtocolMismatchData generate things automatically, but clearly create an array with the pairs that you want to test.